src/Microsoft.DotNet.Core.Build.Tasks/GenerateRuntimeConfigurationFiles.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.DotNet.ProjectModel;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Tasks
+{
+    /// <summary>
+    /// Generates the $(project).runtimeconfig.json and optionally $(project).runtimeconfig.dev.json files
+    /// for a project.
+    /// </summary>
+    public class GenerateRuntimeConfigurationFiles : Task
+    {
+        [Required]
+        public string RuntimeOutputPath { get; set; }
+
+        [Required]
+        public string OutputName { get; set; }
+
+        [Required]
+        public string LockFilePath { get; set; }
+
+        public string RawRuntimeOptions { get; set; }
+
+        public bool IncludeDevConfig { get; set; }
+
+        private LockFile LockFile { get; set; }
+
+        public override bool Execute()
+        {
+            LockFile = LockFileCache.Instance.GetLockFile(LockFilePath);
+
+            WriteRuntimeConfig();
+
+            if (IncludeDevConfig)
+            {
+                WriteDevRuntimeConfig();
+            }
+
+            return true;
+        }
+
+        private void WriteRuntimeConfig()
+        {
+            var json = new JObject();
+            var runtimeOptions = new JObject();
+            json.Add("runtimeOptions", runtimeOptions);
+
+            WriteFramework(runtimeOptions);
+            WriteRuntimeOptions(runtimeOptions);
+
+            var runtimeConfigJsonFile =
+                Path.Combine(RuntimeOutputPath, OutputName + FileNameSuffixes.RuntimeConfigJson);
+
+            using (var writer = new JsonTextWriter(new StreamWriter(File.Create(runtimeConfigJsonFile))))
```

Comment:
The current code doesn't worry about this.  This is copied more or less verbatim from https://github.com/dotnet/cli/blob/feature/msbuild/src/Microsoft.DotNet.Compiler.Common/Executable.cs#L192.

I can't share the code however, because there is no "good" place to refactor it to.  Neither of these assemblies should depend on the other.  And the old code will get deleted and this is the new place for it.  So duplicating the code seemed the right thing to do here.